subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/ModuleResolveState.java

```diff
@@ -387,6 +387,10 @@ public void maybeUpdateSelection(NodeState currentlyDeselecting) {
             // Selection deferred as we know another selector will be added soon
             return;
         }
+        if (selectors.size() == 0) {
```

Comment:
I am really surprised we have to do this check here. Both code path leading there should already confirm that we have one selector at least.
So I wonder if the issue should not be fixed higher in the call stack as we may be doing useless work even before.
Makes sense?

Reply:
Yes. Good point. I moved it up. Wdyt?